evaluation/human_evaluation/main.py

```diff
@@ -0,0 +1,112 @@
+            - **Single values**: Enter individual row numbers separated by commas (e.g., `2,3,4`).
+            - **Range of values**: Use a hyphen to specify a range (e.g., `2-5` translates to `2,3,4,5`).
+            - **Mixed ranges and single values**: Combine both methods separated by commas (e.g., `2-3,4-6` translates to `2,3,4,5,6`).
+            - **Open-ended ranges**: Use a hyphen without an ending number to specify all values starting from a number (e.g., `4-` translates to all values from `4` up to the maximum value in the dataset).
+            - Note: Values starting from `1` are not allowed and will result in an error.
+            """)
+
+        if custom_input:
+            try:
+                parsed_values = parse_custom_input(custom_input, question_count)
+                if 1 in parsed_values:
+                    valid_input = False
+                    st.error("Parsed values cannot include 1.")
+                else:
+                    valid_input = True
+            except ValueError:
+                valid_input = False
+                st.error("Please enter a valid range or list of row numbers.")
+
+    elif selected_option == "All New Questions":
+        parsed_values = find_new_questions(question_count)
+
+    elif selected_option == "All":
+        parsed_values = parse_custom_input("2-", question_count)
+
+button_disabled = (
+    selected_option == ""
+    or (
+        selected_option == "Custom"
+        and not (custom_input and parsed_values and valid_input)
+    )
+    or question_count == 0
+)
+
+if st.button(
+    "Click this button to process (takes time depending on the number of questions)",
+    disabled=button_disabled,
+):
+    if questions:
+        progress = st.progress(0)
+        status_text = st.empty()
+        current_question_text = st.empty()
+
+        with st.spinner("Processing..."):
+            questions_descriptions = read_question_and_description()
+            if not all(isinstance(qd, dict) for qd in questions_descriptions):
+                st.error(
+                    "Invalid data format from Google Sheet. Ensure it has appropriate questions and descriptions."
+                )
+                questions_descriptions = []
+
+            questions_to_process = selected_questions(questions, parsed_values)
+            responses = get_responses(
+                questions_to_process, progress, status_text, current_question_text
+            )
+            updated_cells = write_responses(responses, parsed_values)
+            st.success("Answers generated successfully")
+
+            questions_descriptions = read_question_and_description()
+            update_gform(questions_descriptions)
+            st.success(
+                f"{updated_cells} cells updated successfully! Here is the Google Form: [Google Form Link](https://docs.google.com/forms/d/{GOOGLE_FORM_ID})"
+            )
+    else:
+        st.error("No questions found in the Google Sheet.")
```

Comment:
Use `if __name__ == "__main__"`  to separate main logic from global variables

Reply:
added!